aws-ebs-csi-driver/templates/statefulset.yaml

```diff
@@ -1,26 +1,36 @@
-{{- if .Values.enableVolumeSnapshot }}
-#Snapshot controller
-kind: StatefulSet
-apiVersion: apps/v1
-metadata:
-  name: ebs-snapshot-controller
-  namespace: kube-system
-spec:
-  serviceName: ebs-snapshot-controller
-  replicas: 1
-  selector:
-    matchLabels:
-      app: ebs-snapshot-controller
-  template:
-    metadata:
-      labels:
-        app: ebs-snapshot-controller
-    spec:
-      serviceAccount: ebs-snapshot-controller
-      containers:
-        - name: snapshot-controller
-          image: quay.io/k8scsi/snapshot-controller:v2.0.1
-          args:
-            - --v=5
-            - --leader-election=false
-{{- end }}
+{{- if .Values.enableVolumeSnapshot }}
+#Snapshot controller
```

Comment:
The snapshot controller should be deployed as part of Kubernetes cluster deployment, not the CSI driver. The snapshot controller handles all operations across all CSI drivers.

Reply:
Hi,
I understand that the snapshot controller can be used for various CSI implementation within the same cluster, but in terms of placement within the cluster:

How is that different from the main ebs-csi-controller ?
Both are part of the extended control plane of each cluster and we run them on the controller nodes.
Thx for any clarification ;-)

Reply:
For example, let's say you deploy all these csi drivers in your cluster: ebs, efs, lustre, ceph, xyz, etc.

Even though you have 1+ csi driver in the cluster, only 1 snapshot-controller should be deployed. Therefore snapshot controller deployment needs to live outside of any one csi driver.

Reply:
What @msau42 says makes sense if that how snapshot controller was designed. The situation now is there is no such kubernetes cluster deployment creates the snapshot controller on aws, and we need a place to create the controller

Reply:
It will be great if kops can create it after cluster creation./cc @justinsb

Reply:
In the meantime, can we at least make the snapshot controller deployment optional with clear documentation that you should only have one deployed per cluster?

Reply:
I would also caution against putting the snapshot CRD definitions in any sort of automated driver deployment method. Deletion of a CRD means all the objects also get deleted, which means all the snapshots get deleted.